src/Security/Domain/Authentication/Model/LocalProvider.php

```diff
@@ -256,4 +249,135 @@ public function getProviderRefreshToken(string $token): ?ProviderToken
     {
         return null;
     }
+
+    /**
+     * Check if local security policy is respected
+     *
+     * @param User $user
+     * @param SecurityPolicy $securityPolicy
+     * @param bool $doesPasswordMatch
+     */
+    private function respectLocalSecurityPolicyOrFail(
+        User $user,
+        SecurityPolicy $securityPolicy,
+        bool $doesPasswordMatch,
+    ): void {
+        $isUserBlocked = $this->isUserBlocked($user, $securityPolicy, $doesPasswordMatch);
```

Comment:
```suggestion
        $isUserBlocked = false;
        if ($securityPolicy->getBlockingDuration() !== null) {
            $isUserBlocked = $this->isUserBlocked($user, $securityPolicy, $doesPasswordMatch);
        }
```
Only call this method if blockingDuration is set in policy